src/sentry/workflow_engine/handlers/action/notification/handler.py

```diff
@@ -83,15 +86,77 @@ def execute(
         action: Action,
         detector: Detector,
     ) -> None:
-        try:
-            handler = group_type_notification_registry.get(detector.type)
-            handler.handle_workflow_action(job, action, detector)
-        except NoRegistrationExistsError:
-            logger.exception(
-                "No notification handler found for detector type: %s",
-                detector.type,
-                extra={"detector_id": detector.id, "action_id": action.id},
-            )
+        execute_via_group_type_registry(job, action, detector)
+
+
+@action_handler_registry.register(Action.Type.DISCORD)
+class DiscordActionHandler(NotificationActionHandler):
+    group = NotificationActionHandler.ActionGroup.NOTIFICATION
+
+
+@action_handler_registry.register(Action.Type.SLACK)
+class SlackActionHandler(NotificationActionHandler):
+    group = NotificationActionHandler.ActionGroup.NOTIFICATION
+
+
+@action_handler_registry.register(Action.Type.MSTEAMS)
+class MsteamsActionHandler(NotificationActionHandler):
+    group = NotificationActionHandler.ActionGroup.NOTIFICATION
```

Comment:
```suggestion
    group = ActionHandler.Group.NOTIFICATION
```
^ we should probably access this from the top level `ActionHandler` -- that way it's a little clearer that this is defined a little higher up the OOP stack.